src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -20,25 +20,42 @@ public class DeleteCommand extends Command {
     public static final String COMMAND_WORD = "delete";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Deletes the student identified by the Student ID used in the displayed person list.\n"
+            + ": Deletes the student identified by the Student ID used in the displayed person list,\n"
+            + "or deletes a module from the person's module list.\n"
             + "Parameters: "
-            + PREFIX_STUDENTID + "ID\n"
+            + "ID\n"
+            + "or: "
+            + "ID MODULE_KEYWORD"
             + "Example: " + COMMAND_WORD + " "
-            + PREFIX_STUDENTID + "12345678";
+            + "12345678"
+            + "or: " + COMMAND_WORD + " "
+            + "12345678 m/CS2103T";
 
     public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Student: %1$s";
+    public static final String MESSAGE_DELETE_MODULE_SUCCESS = "Deleted Module: %1$s";
+
     public static final String MESSAGE_PERSON_NOT_FOUND = "No student is found with Student ID: %1$s";
+    public static final String MESSAGE_MODULE_NOT_FOUND = "No module is found for this student: %1$s";
     private final StudentId studentId;
+    private final Module module;
 
     /**
      * Creates a DeleteCommand to delete the person identified by the specified {@code StudentId}.
-     *
-     * @param studentId The student ID of the person to be deleted.
-     * @throws NullPointerException if the {@code studentId} is null.
      */
     public DeleteCommand(StudentId studentId) {
         requireNonNull(studentId);
         this.studentId = studentId;
+        this.module = null;
+    }
+
+    /**
+     * Creates a DeleteCommand to delete the module by the specified {@code Module}.
+     */
+    public DeleteCommand(StudentId studentId, Module module) {
+        requireNonNull(studentId);
+        requireNonNull(module);
```

Comment:
The use of requireNonNull for both studentId and module is a good practice to prevent null pointer exceptions.